### PR TITLE
Prepare major release and allow symfony 6

### PR DESCRIPTION
**Pull request by acrobat**
| Q             | A
| ------------- | ---
| Branch?       | "master" for new features / the branch of the current release for fixes
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | 
| License       | MIT
| Doc PR        |

This PR adds some required typehints for symfony 6 and to make this change easier I suggest to release this PR as a new major release. It's easy for project to upgrade/support the upcoming major, as only a previous deprecation is removed and all older symfony versions are still supported (+ sf6 support)

### CODE REVIEW

**Comment by acrobat** [comment]
All tests pass again, the PR is ready for review!

File: src/ChainRouteCollection.php [hunk 1]
```diff
@@ -98,14 +98,16 @@ public function all()
      *
      * @return Route|null A Route instance or null when not found
```
Comment (dbu) [inline src/ChainRouteCollection.php:99]:
```suggestion
```

File: src/ChainRouter.php [hunk 2]
```diff
@@ -213,23 +210,13 @@ private function doMatch($pathinfo, Request $request = null)
...
+     * @param string $name
      *
      * Loops through all registered routers and returns a router if one is found.
      * It will always return the first route generated.
      */
-    public function generate($name, $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH)
+    public function generate($name, $parameters = [], $absolute = UrlGeneratorInterface::ABSOLUTE_PATH): string
     {
-        if (is_object($name)) {
-            @trigger_error('Passing an object as route name is deprecated since version 2.3. Pass the `RouteObjectInterface::OBJECT_BASED_ROUTE_NAME` as route name and the object in the parameters with key `RouteObjectInterface::ROUTE_OBJECT`.', E_USER_DEPRECATED);
```
Comment (dbu) [inline src/ChainRouter.php:230]: should we check $name with is_string and error if it is not a string?

**Review by dbu — COMMENTED** [review]
thanks for that work!

you are removing the deprecation about passing objects - i think if we don't have `string` parameter type declarations, we should check with is_string to have clean error reporting rather than end up in unexpected behaviour.

**Comment by acrobat** [comment]
I've add the necessary type checks and tests for this case

File: src/ChainRouter.php (hunk 2 above)
Comment (acrobat) [reply src/ChainRouter.php:230]: Good point, I've applied the check!

**Review by dbu — APPROVED** [review]
thanks for the work!

**Comment by dbu** [comment]
i adjusted the branch alias in composer.json.

can you add a build for php 8.1 as well? that might lead to further changes because of deprecations.

**Pull request merged**